lib/fluent/plugin/out_mongo.rb

```diff
@@ -213,23 +214,30 @@ def collect_records(chunk)
         if date_keys
           date_keys.each { |date_key|
             begin
-              date_value = record[date_key]
+              date_key_accessor = record_accessor_create(date_key)
```

Comment:
`record_accessor_create` should be called in `configure`.
Call heavy method in `write` causes slow operation.

Reply:
I have implemented the fix you proposed. Now during the configuration a Hash `accessors` is created with all the data_keys passed as argument